doc/design/draft-metadata-migration.md

```diff
@@ -0,0 +1,29 @@
+Premise: Draft metadata should always be in the latest schema version.
+
+There are a few places in the API where draft metadata are altered (POST, PUT) and served (GET). Note that the UI and
+API use a shared schema version. All actions should thus assume that this is the only version available to the UI and
+server. Any other versions need to be migrated to this version for any display or editing.
+
+This proposal only applies to the `schemaVersion` of dandiset metadata, not asset metadata.
+
+1. The API should provide `GET /dandisets/{dandiset_pk}/versions/{version}/migrated/` endpoint, which would return the migrated metadata of the version, and a description of changes and warnings.
```

Comment:
i'm assuming this always migrates to the API schema version that is deployed. there is no option to migrate to some other schema version. perhaps worth a clarification.

Reply:
Correct. Supporting arbitrary versions would be possible, but significantly more work and it's not something we need right now. I will clarify.